src/amici.cpp

```diff
@@ -103,15 +103,7 @@ std::unique_ptr<ReturnData> runAmiciSimulation(Solver &solver, const ExpData *ed
         rdata->invalidate(model.t0());
         rdata->status = AMICI_ERROR;
         amici::warnMsgIdAndTxt("AMICI:mex:simulation","AMICI simulation failed:\n%s\nError occured in:\n%s",ex.what(),ex.getBacktrace());
-    } catch (std::exception const& ex) {
-        rdata->invalidate(model.t0());
-        rdata->status = AMICI_ERROR;
-        amici::warnMsgIdAndTxt("AMICI:mex:simulation","AMICI simulation failed:\n%s",ex.what());
-    } catch (...) {
-        rdata->invalidate(model.t0());
-        rdata->status = AMICI_ERROR;
-        amici::warnMsgIdAndTxt("AMICI:mex", "Unknown internal error occured");
-    }
+    } 
     model.setFixedParameters(originalFixedParameters);
```

Comment:
Hmm..  this leaves the model in an unexpected state. I guess we should catch, revert to original values and rethrow?
Otherwise this might lead to some surprises if you reuse the model after catching.